### ISSUE TITLE
A daring journey to the bottom of the pit

### ISSUE THREAD

**Issue by mcwill97** [dariusk/NaNoGenMo-2015#98]
I'm going to try my hardest. I really like the idea of writing a book through code! Hopefully it won't be a drawn on bore to read.
COMPLETED EDIT:
Alright so I have absolutely no idea how this site works at all. I hope linking to pastebin will be alright? To be honest I'm not entirely sure how pastebin works and I could be using both these sites entirely wrong. hope not though!
So I made it in unity because I'm used to using unity and I'm currently making a game in it so it was just most convenient for me.

The main script file-
http://pastebin.com/SgSZFWYW
The Character script file-
http://pastebin.com/GG1Ap4kx
The actual mess of repetitive nonsense it spewed out-
http://pastebin.com/qrMYJUzS

I truly fear for anyone who looks through the code. I've never had any actual programmers look at my code. I've self taught myself everything I know about coding, so I could be doing everything terribly. I do know though that there are probably quite a few bugs in that code and also I know its done super messily. Just wanted to get it done after awhile. Anyways, I hope whoever looks at it enjoys it, cause I had fun making it!

**Comment by cpressey** [comment]
Welcome!  I think you'll find that writing code that writes a book that is not-boring to read for the first few hundred words is not too difficult.

After those first few hundred words, though... well, all I can suggest is you download one of the completed novels (from this year or from any earlier year) and try to read the whole thing.  The word "boring" does not quite do justice to the experience.

**Comment by tra38** [comment]
Many entries this year are side-stepping the 'boredom' problem by coming up with a "frame story" to justify essentially an anthology of short stories. Each short story would (theoretically) be coherent and interesting, and so the larger work may stand on its own as being sustainable. For example, some of the proposed entries are 'travelogues', and each 'story' would deal with the main characters visiting a location and interacting with it before moving onto the next scene. Last year, people generated cookbooks and dictionaries as well.

**Comment by cpressey** [comment]
Yes, but I don't think the "many small works" strategy is actually very successful at side-stepping the "boredom problem"... if anything, it might be even harder to read all the way through a list of 500 generated recipes, than through a single generated story of the same length.

**Comment by mcwill97** [comment]
My goal is to write a story in segments with an opening and then a few small arks in the story that hold a main goal for my randomly generated characters to travel through with randomly generated tasks and events

**Comment by enkiv2** [comment]
I agree. Many small works does, however, counteract the problem of trying
to bake in internal consistency and continuity -- and keeping consistency
and continuity forces generators to be simplified or constrained in ways
that can produce more boring text at novel length. So, having an anthology
format can indirectly help in preventing the novel from being boring or
formulaic, by explaining away the existence of novelty sources that outside
an anthology format might be seen as confusing or an error (in the same way
that explicitly nonchronological stories, extremely abstract and vague
prose, swapping out narrators, or having unreliable narrators can).

**Comment by hugovk** [comment]
Many small works taken to the extreme in https://github.com/dariusk/NaNoGenMo-2015/issues/78:
https://hugovk.github.io/NaNoGenMo-2015/8334/8334.html

**Comment by tra38** [comment]
> Anyways, I hope whoever looks at it enjoys it, cause I had fun making it!

I enjoyed it, though said enjoyment did require me to skim through the entire text. That's usually on par with most NaNoGenMo simulated works though. Good job on your first program. May you make others in the future.

**Comment by hugovk** [comment]
Linking to Pastebin is totally fine, we're only using GitHub as a forum. I can see the code files and book so looks like you figured out Pastebin :)

No-one here minds bad code, there'll be plenty being written in this short month. Anyway, code looks fine on a quick skim, and the output is what matters, and I like it:

> You're attacked by a warlock! More than one!
>  A warlock hit The long haired one! It was a super weak attack.
>  The long haired one killed the warlock with power! A warlock hit The deep sounding one!
>  It was a super weak attack. The deep sounding one killed the warlock with wit!
>  The enemies are gone now. What a fight!

I like the repetition too:

> The long haired one thinks she sees people! The long haired one is worried they might be bad people.
>  The deep sounding one thinks the group should try and talk to them. Our group leaves the people alone.
>  The long haired one thinks she sees people! The long haired one is worried they might be bad people.
>  The deep sounding one thinks the group should try and talk to them. Our group leaves the people alone.
>  The long haired one thinks she sees people! The long haired one is worried they might be bad people.
>  The deep sounding one thinks the group should try and talk to them. Our group heads towards the people.
>  Our group asks the people if they know anything about the pit. They ask Our group if we're sure we want to know.
>  Our group says yes. If you really must know, they say, I know the pit is off far away in that direction.
>  A few of them point off in the distance. Our group thanks them for the help and walks where they pointed.
>  What nice people!

**Comment by mattfister** [comment]
This is great - nice job!

**Comment by cpressey** [comment]
I like this a lot!  Good use of tropes.  I wasn't bored by it at all!

**Comment by dariusk** [comment]
Nice one!

**Comment by mcwill97** [comment]
Thanks everyone! I'm really glad you liked it! There's a ton of features I was going to add like more tests that check for the characters stats and repercussions other than death like limb loss and mutations, but it would've lead to a few more days of coding and I was a bit anxious to get back to coding my game since its nearing completion. I am really happy you guys liked it though, because this is the first time I've shown any of my work publicly. In particular I really like what happens when someone likes somebody else who hates them and they try to talk. ctrl - f for "heck" if interested
